### PR TITLE
chef-solo: json_string to HCL2 templates compatibility

### PR DESCRIPTION
**Pull request by sylviamoss**
This adds a new attribute to allow json as string for hcl2 templates.
Closes https://github.com/hashicorp/packer/issues/10532

### CODE REVIEW

**Review by azr — COMMENTED** [review]
Nice one, before I review more thoroughly, do you think it would be a good idea to add a test case for that one ?

File: provisioner/chef-solo/provisioner.go [hunk 1]
```diff
@@ -45,26 +47,27 @@ var guestOSTypeConfigs = map[string]guestOSTypeConfig{
...
+	ConfigTemplate             string               `mapstructure:"config_template"`
+	CookbookPaths              []string             `mapstructure:"cookbook_paths"`
+	RolesPath                  string               `mapstructure:"roles_path"`
+	DataBagsPath               string               `mapstructure:"data_bags_path"`
+	EncryptedDataBagSecretPath string               `mapstructure:"encrypted_data_bag_secret_path"`
+	EnvironmentsPath           string               `mapstructure:"environments_path"`
+	ExecuteCommand             string               `mapstructure:"execute_command"`
+	InstallCommand             string               `mapstructure:"install_command"`
+	RemoteCookbookPaths        []string             `mapstructure:"remote_cookbook_paths"`
+	Json                       map[string]cty.Value `mapstructure:"json"`
```
Comment (azr) [inline provisioner/chef-solo/provisioner.go:61]: I wonder what happens here for JSON templates 🤔

Comment (sylviamoss) [reply provisioner/chef-solo/provisioner.go:61]: I'm still in the process of testing and writing the tests! I removed the wip but I found an issue and I'm working on it.

File: provisioner/chef-solo/provisioner.go [hunk 2]
```diff
@@ -56,13 +56,18 @@ type Config struct {
...
 	InstallCommand             string   `mapstructure:"install_command"`
 	RemoteCookbookPaths        []string `mapstructure:"remote_cookbook_paths"`
-	Json                       map[string]interface{}
-	PreventSudo                bool     `mapstructure:"prevent_sudo"`
-	RunList                    []string `mapstructure:"run_list"`
-	SkipInstall                bool     `mapstructure:"skip_install"`
-	StagingDir                 string   `mapstructure:"staging_directory"`
-	GuestOSType                string   `mapstructure:"guest_os_type"`
-	Version                    string   `mapstructure:"version"`
+	// HCL cannot decode into interface so we add an option to write the JSON as string.
```
Comment (SwampDragons) [inline provisioner/chef-solo/provisioner.go:59]:
```suggestion
	// HCL cannot be decoded into an interface so for HCL templates you must use the JsonString option,
```

**Review by SwampDragons — APPROVED** [review]
This looks like the right approach. There's no way for a builder to know whether it was originally json or HCL, is there? 

Another thought -- we should fix hcl2_upgrade to handle this case using the json_encode func

**Comment by sylviamoss** [comment]
Currently, the builder doesn't know the config type but this is possible to get if we add that info to the PackerConfig 🤔  
We can think about that we find it necessary.

**Comment by azr** [comment]
Nice one, LGTM; I think the approach of going for a string is ok !

**Pull request merged**